sympy/stats/tests/test_discrete_rv.py

```diff
@@ -217,9 +217,14 @@ def test_conditional():
 def test_product_spaces():
     X1 = Geometric('X1', S(1)/2)
     X2 = Geometric('X2', S(1)/3)
-    assert str(P(X1 + X2 < 3, evaluate=False)) == """Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, """\
-        + """(-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, -oo, -1))"""
+    #assert str(P(X1 + X2 < 3, evaluate=False)) == """Sum(Piecewise((2**(X2 - n - 2)*(2/3)**(X2 - 1)/6, """\
+    #    + """(-X2 + n + 3 >= 1) & (-X2 + n + 3 < oo)), (0, True)), (X2, 1, oo), (n, -oo, -1))"""
+    assert str(P(X1 + X2 < 3, evaluate=False)) == 'Sum(Piecewise((2**(-n)/4,'\
```

Comment:
It seems like the evaluate flag is not propagated here. I cannot really understand why since I didn't touch anything related to that (that I know of). Apart from that, the test should be fine, I think (inheriting assumptions means that the Piecewise relationals are simplified).

Reply:
If evaluate is False then is `doit=False` being sent to `simplify` when called somewhere on P's path?

Reply:
It seems a bit weird that simplify should be called upon instantiation. I have not really considered it more than making that statement to myself...

Reply:
This code should be using Dummys in the sum. We can get rid of the str test by using:
```julia
In [118]: s1 = P(X1 + X2 < 3, evaluate=False)                                                            

In [119]: s2 = Sum(Piecewise((2**(X2 - n - 2)*(S(2)/3)**(X2 - 1)/6, (-X2 + n + 3 >= 1) & (-X2 + n + 3 < o
     ...: o)), (0, True)), (X2, 1, oo), (n, -oo, -1))                                                    

In [120]: s1 = s1.replace(X2.symbol, Dummy('X2', integer=True)).replace(Symbol('n'), Dummy('n', integer=T
     ...: rue))                                                                                          

In [121]: s2 = s2.replace(X2, Dummy('X2', integer=True)).replace(Symbol('n', integer=True), Dummy('n', in
     ...: teger=True))                                                                                   

In [122]: s1.dummy_eq(s2)                                                                                
Out[122]: True
```

Reply:
Agreed. These things showed up at a stage where I had put many more hours than intended on it and since I wasn't sure if I had the correct result, I didn't bother to change it.

Reply:
I can of course change it, once I am reasonably sure that the results are correct (and not just what happened to come out after the code change...).

Reply:
Fixed these now and hope that they are correct.